website/docs/design-patterns/component-catalog-template.md

```diff
@@ -0,0 +1,194 @@
+        service_account_namespace: { { .service_account_namespace } }
+        # Example of using the Sprig functions in `Go` templates.
+        # Refer to https://masterminds.github.io/sprig for more details.
+        { { if hasKey . "iam_managed_policy_arns" } }
+        iam_managed_policy_arns:
+          { { range $i, $iam_managed_policy_arn := .iam_managed_policy_arns } }
+          - '{{ $iam_managed_policy_arn }}'
+          { { end } }
+        { { - end } }
+```
+
+Import the `stacks/catalog/eks/iam-role/defaults.tmpl` manifest template into a top-level stack,
+for example `stacks/orgs/acme/plat/prod/us-east-2.yaml`, and provide the configuration for each application in the `context` object:
+
+```yaml title="stacks/orgs/acme/plat/prod/us-east-2.yaml"
+import:
+  - orgs/acme/plat/prod/_defaults
+  - mixins/region/us-east-2
+
+  # This import will dynamically generate a new Atmos component `eks/iam-role/admin-ui`
+  - path: catalog/eks/iam-role/defaults.tmpl
+    context:
+      app_name: "admin-ui"
+      service_account_name: "admin-ui"
+      service_account_namespace: "admin"
+      iam_managed_policy_arns: [ "<arn1>", "<arn2>" ]
+
+  # This import will dynamically generate a new Atmos component `eks/iam-role/auth`
+  - path: catalog/eks/iam-role/defaults.tmpl
+    context:
+      app_name: "auth"
+      service_account_name: "auth"
+      service_account_namespace: "auth"
+      iam_managed_policy_arns: [ "<arn3>" ]
+
+  # This import will dynamically generate a new Atmos component `eks/iam-role/payment-processing`
+  - path: catalog/eks/iam-role/defaults.tmpl
+    context:
+      app_name: "payment-processing"
+      service_account_name: "payment-processing"
+      service_account_namespace: "payments"
+      iam_managed_policy_arns: [ "<arn4>", "<arn5>" ]
+
+  # Add new application configurations here
+```
+
+To provision the Atmos components in the stack, execute the following commands:
+
+```shell
+atmos terraform apply eks/iam-role/admin-ui --stack plat-ue2-prod
+atmos terraform apply eks/iam-role/auth -s plat-ue2-prod
+atmos terraform apply eks/iam-role/payment-processing -s plat-ue2-prod
+```
+
+## Benefits
+
+The **Component Catalog Template** pattern provides the following benefits:
+
+- All settings for a component are defined in just one place (in the component's template) making the entire
+  configuration [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
+
+- Many instances of the component can be provisioned without repeating all the configuration values
+
+- New Atmos components are generated dynamically
+
+## Limitations
+
+The **Component Catalog Template** pattern has the following limitations and drawbacks:
+
+- Since new Atmos components are generated dynamically, sometimes it's not easy to know the names of the Atmos components that need to be provisioned
+  without looking at the `Go` template and figuring out all the Atmos component names
+
```

Comment:
```suggestion
- Since new Atmos components are generated dynamically, sometimes it's not easy to know the names of the Atmos components that need to be provisioned
  without looking at the `Go` template and figuring out all the Atmos component names
- With great power comes great responsibility. The Go templating engine leads to infinite possibilities, which makes the stack configs more challenging to maintain and reduces consistency. Try to leverage the inheritance model as much as possible, over templated stack configs.
```